domain/src/test/java/com/thoughtworks/go/helper/GitRepoContainingSubmodule.java

```diff
@@ -131,7 +97,9 @@ public String projectRepositoryUrl() {
         changeFile(remoteSubmoduleRepoLocation, fileName, newContentOfFile);
         checkInOneFile(remoteSubmoduleRepoLocation, new File(fileName), comment);
 
-        git(new File(remoteRepoDir, submoduleNameInRepo)).pull();
+        CommandLine.createCommandLine("git").withEncoding("UTF-8").withArg("pull").
+                withWorkingDir(new File(remoteRepoDir, submoduleNameInRepo)).
+                runOrBomb(new MaterialFingerprintTag(null));
```

Comment:
This is just some refactoring to get rid of `GitCommand#pull()`, which was a `@TestOnly` method. The rest is IDEA cleanup/method re-org. This is the only method I removed because I got too lazy to factor out the other `@TestOnly` methods. I also figured that effort is best saved for a separate PR.